File: sorting/selection/selection-sort.cpp
#include <iostream>
using namespace std;

void print(int st_bf[], size_t size) {
    for (size_t i = 0; i < size; i++)
    {
        cout<< st_bf[i] << " ";
    }
    cout<< endl;
}

void select_sort(int *st_bf, size_t size) {
    // 选择排序
    for (size_t i = 0; i < size - 1; i++) {
        int temp = st_bf[i]; // 缓存需要比较的值
        size_t location = i; // 缓存最小值的位置
        for (size_t j = i + 1; j < size; j++) {
            // 找出剩余部分最小值
            if (st_bf[j] < temp) {
                temp = st_bf[j];
                location = j;
            }
        }
        if (i != location) {
            // 交换当前位置数和最小数
            st_bf[location] = st_bf[i];
            st_bf[i] = temp;
        }
        print(st_bf, size);
    }
}

int main() {
    int st_bf[] = {6, 5, 3, 1, 8, 7, 2, 4, 2};
    size_t size = sizeof(st_bf) / sizeof(st_bf[0]);
    select_sort(st_bf, size);
}

File: sorting/backups/selection_sort.py
# -*- encoding: utf-8 -*-


def select_sort(st_bf):
    length = len(st_bf)
    for i in xrange(length - 1):
        temp, location = st_bf[i], i

        # 找出剩余部分最小值
        for j in xrange(i+1, length):
            if st_bf[j] < temp:
                temp, location = st_bf[j], j

        if location != i:
            st_bf[location], st_bf[i] = st_bf[i], st_bf[location]
        print st_bf


st_bf = [6, 5, 3, 1, 8, 7, 2, 4, 2]
select_sort(st_bf)
